test/unit/spec/common/jobManager-Spec.js

```diff
@@ -818,14 +844,31 @@ define([
                     expect(
                         this.jobManagerInstance.model.getItem(`exec.jobs.info.${jobId}`)
                     ).not.toBeDefined();
-                    // rather than faffing around with sending messages over the bus,
-                    // trigger the job handler directly
-                    this.jobManagerInstance.runHandler(jcm.MESSAGE_TYPE.INFO, jobInfo, jobId);
 
-                    expect(
-                        this.jobManagerInstance.model.getItem(`exec.jobs.info.${jobId}`)
-                    ).toEqual(jobInfo);
-                    expect(this.jobManagerInstance.listeners[jobId][event]).not.toBeDefined();
+                    return new Promise((resolve) => {
+                        // add a handler that acts after the default handler so we know
+                        // the default action has already been done
+                        this.jobManagerInstance.addListener(
+                            jcm.MESSAGE_TYPE.INFO,
+                            jcm.CHANNEL.JOB,
+                            jobId,
+                            {
+                                zzz_resolve_promise: (...args) => {
+                                    // eslint-disable-next-line no-unused-vars
+                                    const [_, msg, channel] = args;
+                                    expect(msg).toEqual(jobInfo);
+                                    expect(channel).toEqual(channelData);
+                                    resolve();
+                                },
```

Comment:
added a handler that should act last and resolve a promise -- otherwise the test fails as there is a slight lag in processing messages coming over the bus